app/services/plus_one.rb

```diff
@@ -0,0 +1,28 @@
+class PlusOne
+  class CannotPlusOneYourself < StandardError; end
+  class InvalidSlackToken < StandardError; end
+
+  def initialize(team)
+    @team = team
+  end
+
+  def call(params)
+    ActiveRecord::Base.transaction do
+      recipient_name = MessageParser.new(params[:text], params[:trigger_word]).recipient_name
+      sender, recipient = prepare_transaction_actors.(params)
+      raise InvalidSlackToken if recipient.slack_user_name == 'u'
```

Comment:
it should do the job in most cases, but it relies on something that can change - why is the magical 'u' here? is that the only way we know that something went wrong?